sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/ReviveManager.java

```diff
@@ -1,81 +1,37 @@
 import com.mesosphere.sdk.scheduler.plan.PodInstanceRequirement;
+import com.mesosphere.sdk.scheduler.plan.Status;
 import com.mesosphere.sdk.scheduler.plan.Step;
-import com.mesosphere.sdk.state.StateStore;
-import com.mesosphere.sdk.state.StateStoreUtils;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
 import org.apache.mesos.SchedulerDriver;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.util.Collection;
+import java.util.HashSet;
+import java.util.Optional;
 import java.util.Set;
-import java.util.concurrent.Executors;
-import java.util.concurrent.ScheduledExecutorService;
-import java.util.concurrent.TimeUnit;
 import java.util.stream.Collectors;
 
 /**
- * This class monitors a {@link PlanCoordinator} and revives offers when appropriate.
+ * This class determines whether offers should be revived based on changes to the work being processed by the scheduler.
  */
 public class ReviveManager {
-    public static final int REVIVE_INTERVAL_S = 5;
-    public static final int REVIVE_DELAY_S = 5;
 
     private final Logger logger = LoggerFactory.getLogger(getClass());
     private final SchedulerDriver driver;
-    private final PlanCoordinator planCoordinator;
-    private final ScheduledExecutorService monitor = Executors.newScheduledThreadPool(1);
-    private final StateStore stateStore;
-    private Set<PodInstanceRequirement> candidates;
-
-    public ReviveManager(
-            SchedulerDriver driver,
-            StateStore stateStore,
-            PlanCoordinator planCoordinator) {
-        this(
-                driver,
-                stateStore,
-                planCoordinator,
-                REVIVE_DELAY_S,
-                REVIVE_INTERVAL_S);
-    }
-
-    public ReviveManager(
-            SchedulerDriver driver,
-            StateStore stateStore,
-            PlanCoordinator planCoordinator,
-            int pollDelay,
-            int pollInterval) {
+    private final TokenBucket tokenBucket;
+    private Set<WorkItem> candidates = new HashSet<>();
 
-        this.driver = driver;
-        this.stateStore = stateStore;
-        this.planCoordinator = planCoordinator;
-        this.candidates = getRequirements(planCoordinator.getCandidates());
-        monitor.scheduleAtFixedRate(
-                new Runnable() {
-                    @Override
-                    public void run() {
-                        revive();
-                    }
-                },
-                pollDelay,
-                pollInterval,
-                TimeUnit.SECONDS);
-
-        logger.info(
-                "Monitoring these plans for suppress/revive: {}",
-                planCoordinator.getPlanManagers().stream()
-                        .map(planManager -> planManager.getPlan().getName())
-                        .collect(Collectors.toList()));
+    public ReviveManager(SchedulerDriver driver) {
```

Comment:
IMO, ReviveManager should take the TokenBuilder as a dependency, that would allow adding tests of the critical switch around token fulfillment.

Reply:
Alternately, you could just have a second constructor that takes it as a parameter, then only use that directly in tests (e.g. make it protected).

Reply:
There is a second constructor immediately below this comment.  Am I missing something?